tapdb/sqlc/queries/assets.sql

```diff
@@ -854,7 +854,14 @@ INSERT INTO script_keys (
 )  ON CONFLICT (tweaked_script_key)
     -- As a NOP, we just set the script key to the one that triggered the
     -- conflict.
-    DO UPDATE SET tweaked_script_key = EXCLUDED.tweaked_script_key
+    DO UPDATE SET 
+      tweaked_script_key = EXCLUDED.tweaked_script_key,
+      -- If the script key was previously unknown, we'll update to the new
+      -- value.
+      declared_known = CASE
+                         WHEN script_keys.declared_known = FALSE THEN EXCLUDED.declared_known
```

Comment:
Opted for this case statement to make sure that if the ordering gets reversed, we don't end up making something as unknown after it was marked as known.

Reply:
Shouldn't this be `WHEN script_keys.declared_known != TRUE`? Otherwise we won't set it when it's `NULL`.

Reply:
So I wrote a test to try to make things fail before fixing (pushed up now), but but it still passed. I copied some code so I could insert a `NULL` for the bool, but without any change, the extra case still passed.

Reply:
Ah, so we don't allow [it to be null in the first place](https://github.com/lightninglabs/taproot-assets/blob/7c1cd9a32937415d1014bb16b39d4c4512bfa713/tapdb/sqlc/migrations/000018_script_key_declared_known.up.sql#L5), just at times the response after a join can be NULL, and a query might use `sqlc.narg`. Either way I updated to the new version as it's more explicit, and slightly more compact.

Reply:
Nvm, my test was missing a statement 🤦 (fails when that missing statement is added). I also misinterpreted the initial migration: you need to specify `NOT NULL` otherwise it's allowed. 

Updated, and now accounts for `NULL`. Test passes!